### PR TITLE
LightGBM support! ^_^

### PR DESCRIPTION
**Pull request by lavanyashukla**

### CODE REVIEW

File: wandb/lightgbm/__init__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import wandb
+
+
+def wandb_callback():
+    def callback(env):
+        eval_results = {}
+        recorder = lightgbm.record_evaluation(eval_results)
+        recorder(env)
+
+        print(eval_results)
```
Comment (raubitsj) [inline wandb/lightgbm/__init__.py]: is this print debugging that is intended to stay, if you want to have a print in the callback i would suggest a debug option on wandb_callback and formatting the print output

File: wandb/lightgbm/__init__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        for validation_key in eval_results.keys():
+            for key in eval_results[validation_key].keys():
+                wandb.log({
+                    validation_key + "_" + key: eval_results[validation_key][key][0]
```
Comment (raubitsj) [inline wandb/lightgbm/__init__.py]: are validation_key and key always strings?

File: wandb/lightgbm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                }, commit=False)
+        wandb.log({})
```
Comment (nbardy) [inline wandb/lightgbm/__init__.py]: Can you leave a comment to tell that this is for flushing the open commit.

File: wandb/lightgbm/__init__.py (hunk 2 above)
Comment (lavanyashukla) [reply wandb/lightgbm/__init__.py]: Yup, they are the data_name and eval_name from [here]( https://lightgbm.readthedocs.io/en/latest/_modules/lightgbm/callback.html#record_evaluation) respectively.

File: wandb/lightgbm/__init__.py (hunk 1 above)
Comment (lavanyashukla) [reply wandb/lightgbm/__init__.py]: Fixed in [728](https://github.com/wandb/client/pull/728).

File: wandb/lightgbm/__init__.py (hunk 3 above)
Comment (lavanyashukla) [reply wandb/lightgbm/__init__.py]: Fixed in [728](https://github.com/wandb/client/pull/728).

**Pull request merged**